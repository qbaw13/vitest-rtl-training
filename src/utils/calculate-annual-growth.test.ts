import calculateAnnualGrowth from "./calculate-annual-growth";

describe("calculate annual growth", () => {
  it("should return calculated annual growth", () => {
    const actual = calculateAnnualGrowth(345, 4, 0.0645);

    expect(actual).toEqual([345, 367.2525, 390.94028625, 416.15593471312496, 442.9979925021215]);
  });

  it.each`
    amount | numberOfYears | growthRate
    ${0}   | ${1}          | ${1}
    ${1}   | ${0}          | ${1}
    ${1}   | ${1}          | ${0}
  `(
    "should throw an error when amount=$amount, numberOfYears=$numberOfYears, growthRate=$growthRate",
    ({ amount, numberOfYears, growthRate }) => {
      expect(() => {
        calculateAnnualGrowth(amount, numberOfYears, growthRate);
      }).toThrowErrorMatchingSnapshot();
    },
  );
});
