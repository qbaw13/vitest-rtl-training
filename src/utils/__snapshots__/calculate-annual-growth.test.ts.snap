// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`calculate annual growth > should throw an error when amount=+0, numberOfYears=1, growthRate=1 1`] = `"Amount should be positive number, amount=0"`;

exports[`calculate annual growth > should throw an error when amount=1, numberOfYears=+0, growthRate=1 1`] = `"Number of years should be positive number, numberOfYears=0"`;

exports[`calculate annual growth > should throw an error when amount=1, numberOfYears=1, growthRate=+0 1`] = `"Growth rate should be positive number, growthRate=0"`;
